hail/src/main/scala/is/hail/expr/ir/TypeCheck.scala

```diff
@@ -145,7 +142,7 @@ object TypeCheck {
         }
       case x@MakeArray(args, typ) =>
         assert(typ != null)
-        args.map(_.typ).zipWithIndex.foreach { case (x, i) => assert(x == typ.elementType,
+        args.map(_.typ).zipWithIndex.foreach { case (x, i) => assert(x.isOfType(typ.elementType),
```

Comment:
Needed to make Coalesce test pass; the MakeArray PR issued just after this one further removes == in TypeCheck.